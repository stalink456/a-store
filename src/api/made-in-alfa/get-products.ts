import HttpClient from 'utils/httpClient';
import { Method } from 'api/types';
import { ProductsType } from 'store/types';
import { BASE_URL } from '../../constants';

export const getProducts = async (): Promise<ProductsType[]> => {
  const response = await HttpClient.call(
    Method.Get,
    `${BASE_URL}/made-in-alfa`
  );
  return response.data;
};
