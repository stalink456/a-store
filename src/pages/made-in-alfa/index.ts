export { MadeInAlfa as default } from "./made-in-alfa";
