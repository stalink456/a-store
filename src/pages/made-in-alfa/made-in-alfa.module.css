@import "../../css/variables.css";
.made_in {
  padding: 0 15px;
}

.made_in__container {
  position: relative;
  animation: slide-right 1s ease-in-out;
}

.made_in__title {
  font-size: 100px;
}

.made_in__subtitle {
  margin-top: 20px;
  margin-bottom: 75px;
}

@media (max-width: 1065px) {
  .made_in__title {
    font-size: 70px;
  }

  .made_in__subtitle {
    margin-top: 20px;
    margin-bottom: 50px;
  }
}

@media (max-width: 660px) {
  .made_in__title {
    font-size: 50px;
  }
}

@media (max-width: 520px) {
  .made_in__title {
    font-size: 40px;
  }
  .made_in__subtitle {
    font-size: 18px;
    margin: 0;
  }
}

@media (max-width: 480px) {
  .made_in__container {
    justify-content: center;
  }
}
