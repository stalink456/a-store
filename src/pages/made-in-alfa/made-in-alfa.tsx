import React from 'react';
import { Page } from 'components/ui-components/page';
import { Space } from '@alfalab/core-components/space';
import { Typography } from '@alfalab/core-components/typography';
import { ProductsCard } from 'components/products-card';
import { Loading } from 'components/ui-components/loading';
import { useMadeInAlfa } from 'hooks/use-made-in-alfa';

import styles from './made-in-alfa.module.css';

export const MadeInAlfa: React.FC = () => {
  const { products, isLoading } = useMadeInAlfa();

  if (isLoading) {
    return <Loading />;
  }

  return (
    <Page>
      <div className={styles.made_in}>
        <Space
          className={styles.made_in__container}
          direction='vertical'
          fullWidth
          size={5}
        >
          <Space direction='horizontal' size={5} fullWidth>
            <Typography.Title
              className={styles.made_in__title}
              tag='h1'
              view='xlarge'
              weight='bold'
            >
              Сделано в Альфе
            </Typography.Title>
          </Space>
          <Space direction='horizontal'>
            <Typography.Title
              className={styles.made_in__subtitle}
              tag='h6'
              view='small'
              weight='medium'
            >
              Хотим каждую из этих вещей! Себе, родным и друзьям
            </Typography.Title>
          </Space>
          <Space
            className={styles.made_in__products}
            direction='horizontal'
            wrap
            size={0}
          >
            {products.map((value, id) => (
              <ProductsCard {...value} key={id} />
            ))}
          </Space>
        </Space>
      </div>
    </Page>
  );
};
