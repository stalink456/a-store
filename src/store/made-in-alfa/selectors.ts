import { ApplicationState } from "..";

export const madeInAlfaSelector = (state: ApplicationState) => state.madeInAlfa;

export const madeInAlfaProductsSelector = (state: ApplicationState) =>
  madeInAlfaSelector(state).products;
export const madeInAlfaIsLoadingSelector = (state: ApplicationState) =>
  madeInAlfaSelector(state).isLoading;
export const madeInAlfaIsErrorSelector = (state: ApplicationState) =>
  madeInAlfaSelector(state).hasError;
