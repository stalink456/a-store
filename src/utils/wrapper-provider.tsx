import { Store } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

export const wrapperProvider = (store: Store, component: React.ReactNode) => {
  return <Provider store={store}>{component}</Provider>;
};
