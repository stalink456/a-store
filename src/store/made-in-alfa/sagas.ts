import { call, put, takeLatest } from 'redux-saga/effects';
import { getProducts } from 'api/made-in-alfa/get-products';
import { ProductsType } from 'store/types';
import { madeInAlfaActions } from '.';

function* getMadeInAlfaSaga() {
  try {
    const products: ProductsType[] = yield call(getProducts);

    yield put(madeInAlfaActions.success(products));
  } catch (e) {
    yield put(madeInAlfaActions.failure);
  }
}

export function* watchMadeInAlfaSaga() {
  yield takeLatest(madeInAlfaActions.request.type, getMadeInAlfaSaga);
}
