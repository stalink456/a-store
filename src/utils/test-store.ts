import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "redux-saga";
import { madeInAlfaReducer } from "store/made-in-alfa";
import { yourDesignReducer } from "store/your-design";
import { productReducer } from "store/product";
import { cartReducer } from "store/cart";
import { orderReducer } from "store/order";
import { rootSaga } from "../store/root-saga";

const sagaMiddleware = createSagaMiddleware();
const middlewares = [sagaMiddleware];

export const createTestStore = configureStore({
  reducer: {
    madeInAlfa: madeInAlfaReducer,
    yourDesign: yourDesignReducer,
    product: productReducer,
    cart: cartReducer,
    order: orderReducer,
  },
  devTools: true,
  middleware: middlewares,
});

sagaMiddleware.run(rootSaga);