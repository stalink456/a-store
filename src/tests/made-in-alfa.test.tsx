import axios from "axios";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Store } from "@reduxjs/toolkit";
import { MadeInAlfa } from "pages/made-in-alfa/made-in-alfa";
import Product from "pages/product";
import { createTestStore } from "../utils/test-store";
import * as madeinalfa from "./mocks/get-made-in-alfa-products-200.json";
import * as product from "./mocks/get-made-in-alfa-product-200.json";
import { wrapperProvider } from "utils/wrapper-provider";

const mockProducts = madeinalfa["madeinalfa"];
const mockProduct = product["product"];

axios.get = jest.fn();
const mockedAxios = axios as jest.Mocked<typeof axios>;

let store: Store;

describe("MadeInAlfa component", () => {
  beforeEach(() => {
    store = createTestStore;
  });

  it("should render MadeInAlfa component", async () => {
    mockedAxios.get.mockResolvedValue({ mockProducts });
    const { baseElement } = render(wrapperProvider(store, <MadeInAlfa />), {
      wrapper: MemoryRouter,
    });

    await waitFor(() => {
      expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
    });

    expect(baseElement).toMatchSnapshot();
  });

  it("should render MadeInAlfa component with title", async () => {
    render(wrapperProvider(store, <MadeInAlfa />), { wrapper: MemoryRouter });

    await waitFor(() => {
      expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
    });

    await screen.findByText("Сделано в Альфе");
  });

  it("should render MadeInAlfa component with a transition to Product page", async () => {
    mockedAxios.get.mockResolvedValue({ mockProducts });

    render(
      wrapperProvider(
        store,
        <MemoryRouter initialEntries={["/made-in-alfa"]}>
          <Routes>
            <Route path="/made-in-alfa" element={<MadeInAlfa />} />
            <Route path="/product/:id" element={<Product />} />
          </Routes>
        </MemoryRouter>
      )
    );

    await waitFor(() => {
      expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
    });

    fireEvent.click(screen.getByAltText("Рюкзак «Для умных и свободных»"));

    mockedAxios.get.mockResolvedValue({ mockProduct });

    expect(
      await screen.findByText("Рюкзак «Для умных и свободных»")
    ).toBeInTheDocument();
  });

  it("should render MadeInAlfa component with infinity Spinner", async () => {
    mockedAxios.get.mockResolvedValue([]);

    render(wrapperProvider(store, <MadeInAlfa />), { wrapper: MemoryRouter });

    expect(screen.getByTestId("loading")).toBeInTheDocument();
  });

  it("shouldn't render MadeInAlfa component with non-existent image", async () => {
    render(wrapperProvider(store, <MadeInAlfa />), { wrapper: MemoryRouter });

    const image = screen.queryByAltText(
      "non-existent image"
    ) as HTMLImageElement;
    await waitFor(() => expect(image === null).toBeTruthy());
  });

  it("shouldn't render MadeInAlfa component with non-existent description", async () => {
    render(wrapperProvider(store, <MadeInAlfa />), { wrapper: MemoryRouter });

    const result = screen.queryByAltText(
      "non-existent description"
    ) as HTMLImageElement;
    await waitFor(() => expect(result).not.toBeTruthy());
  });
});
