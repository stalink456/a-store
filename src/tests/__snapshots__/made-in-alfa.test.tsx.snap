// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MadeInAlfa component should render MadeInAlfa component 1`] = `
<body>
  <div>
    <div
      class="page"
    >
      <div
        class="made_in"
      >
        <div
          class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as made_in__container"
        >
          <div
            class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
            style="margin-bottom: 5px;"
          >
            <div
              class="space__spaceContainer_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as"
            >
              <div
                class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
              >
                <h1
                  class="typography__component_19zgr made_in__title typography__styrene-xlarge_pxbut typography__bold_pxbut"
                >
                  Сделано в Альфе
                </h1>
              </div>
            </div>
          </div>
          <div
            class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
            style="margin-bottom: 5px;"
          >
            <div
              class="space__spaceContainer_nr3as space__start_nr3as"
            >
              <div
                class="space__spaceItem_nr3as"
              >
                <h6
                  class="typography__component_19zgr made_in__subtitle typography__styrene-small_pxbut typography__medium_pxbut"
                >
                  Хотим каждую из этих вещей! Себе, родным и друзьям
                </h6>
              </div>
            </div>
          </div>
          <div
            class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
          >
            <div
              class="space__spaceContainer_nr3as space__start_nr3as made_in__products"
              style="flex-wrap: wrap; margin-bottom: 0px;"
            >
              <div
                class="space__spaceItem_nr3as"
                style="margin-right: 0px; padding-bottom: 0px;"
              >
                <div
                  class="products_card"
                >
                  <a
                    href="/product/0"
                  >
                    <div
                      class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
                      style="flex-wrap: wrap; margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <div
                              class="skeleton__component_1bshy skeleton__animate_1bshy"
                            >
                              <img
                                alt="Рюкзак «Для умных и свободных»"
                                class=""
                                src="http://qa-games.ru/astore/public/images/15932051.jpeg"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <h6
                              class="typography__component_19zgr typography__styrene-xsmall_pxbut typography__medium_pxbut"
                            >
                              Рюкзак «Для умных и свободных»
                            </h6>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <p
                              class="typography__paragraph_c16y5 typography__primary-large_c16y5 typography__bold_c16y5"
                              data-testid="price-0"
                            >
                              4999
                               ₽
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
              <div
                class="space__spaceItem_nr3as"
                style="margin-right: 0px; padding-bottom: 0px;"
              >
                <div
                  class="products_card"
                >
                  <a
                    href="/product/1"
                  >
                    <div
                      class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
                      style="flex-wrap: wrap; margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <div
                              class="skeleton__component_1bshy skeleton__animate_1bshy"
                            >
                              <img
                                alt="Футболка Для умных и свободных"
                                class=""
                                src="http://qa-games.ru/astore/public/images/68519498.jpeg"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <h6
                              class="typography__component_19zgr typography__styrene-xsmall_pxbut typography__medium_pxbut"
                            >
                              Футболка Для умных и свободных
                            </h6>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <p
                              class="typography__paragraph_c16y5 typography__primary-large_c16y5 typography__bold_c16y5"
                              data-testid="price-1"
                            >
                              1999
                               ₽
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
              <div
                class="space__spaceItem_nr3as"
                style="margin-right: 0px; padding-bottom: 0px;"
              >
                <div
                  class="products_card"
                >
                  <a
                    href="/product/2"
                  >
                    <div
                      class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
                      style="flex-wrap: wrap; margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <div
                              class="skeleton__component_1bshy skeleton__animate_1bshy"
                            >
                              <img
                                alt="Блокнот Для умных и свободных"
                                class=""
                                src="http://qa-games.ru/astore/public/images/77117755.jpeg"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <h6
                              class="typography__component_19zgr typography__styrene-xsmall_pxbut typography__medium_pxbut"
                            >
                              Блокнот Для умных и свободных
                            </h6>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <p
                              class="typography__paragraph_c16y5 typography__primary-large_c16y5 typography__bold_c16y5"
                              data-testid="price-2"
                            >
                              1499
                               ₽
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
              <div
                class="space__spaceItem_nr3as"
                style="margin-right: 0px; padding-bottom: 0px;"
              >
                <div
                  class="products_card"
                >
                  <a
                    href="/product/3"
                  >
                    <div
                      class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
                      style="flex-wrap: wrap; margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <div
                              class="skeleton__component_1bshy skeleton__animate_1bshy"
                            >
                              <img
                                alt="Чехол с кардхолдером"
                                class=""
                                src="http://qa-games.ru/astore/public/images/15932051.jpeg"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <h6
                              class="typography__component_19zgr typography__styrene-xsmall_pxbut typography__medium_pxbut"
                            >
                              Чехол с кардхолдером
                            </h6>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <p
                              class="typography__paragraph_c16y5 typography__primary-large_c16y5 typography__bold_c16y5"
                              data-testid="price-3"
                            >
                              799
                               ₽
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
              <div
                class="space__spaceItem_nr3as"
                style="padding-bottom: 0px;"
              >
                <div
                  class="products_card"
                >
                  <a
                    href="/product/4"
                  >
                    <div
                      class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
                      style="flex-wrap: wrap; margin-bottom: 0px;"
                    >
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <div
                              class="skeleton__component_1bshy skeleton__animate_1bshy"
                            >
                              <img
                                alt="Экоручка"
                                class=""
                                src="http://qa-games.ru/astore/public/images/56369345.jpeg"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <h6
                              class="typography__component_19zgr typography__styrene-xsmall_pxbut typography__medium_pxbut"
                            >
                              Экоручка
                            </h6>
                          </div>
                        </div>
                      </div>
                      <div
                        class="space__spaceItem_nr3as"
                      >
                        <div
                          class="space__spaceContainer_nr3as space__start_nr3as"
                        >
                          <div
                            class="space__spaceItem_nr3as"
                          >
                            <p
                              class="typography__paragraph_c16y5 typography__primary-large_c16y5 typography__bold_c16y5"
                              data-testid="price-4"
                            >
                              99
                               ₽
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
